dkg/frost/messages.go

```diff
@@ -0,0 +1,41 @@
+package frost
+
+import "encoding/json"
+
+type ProtocolMsg struct {
+	Round DKGRound
+
+	Data interface{}
+}
```

Comment:
I wonder whether it's better  to have the following structure.
 ```go
type ProtocolMsg struct {
	Preparation  *PreparationMessage `json:"preparation,omitempty"`
	Round1       *Round1Message      `json:"round1,omitempty"`
	Round2       *Round2Message      `json:"round2,omitempty"`
	BlameMessage *BlameMessage       `json:"blame,omitempty"`
}
```
Some of the benefits are:
1. We have explicit message type instead of encoding as integer which might be confusing.
2. The initial SSV style with `type` and `data` is so that it's easy to support new types, in our case we don't have new types (if we do, it's better to create a new struct parallel to `ProtocolMsg`).

Reply:
This works too, should I proceed to change it to this type? It will save us a check on casting from interface to explicit type

Reply:
Yes. We can change to this.